lib/plugins/aws/package/compile/events/alexaSkill/index.js

```diff
@@ -15,44 +15,74 @@ class AwsCompileAlexaSkillEvents {
   compileAlexaSkillEvents() {
     this.serverless.service.getAllFunctions().forEach((functionName) => {
       const functionObj = this.serverless.service.getFunction(functionName);
+      let alexaSkillNumberInFunction = 0;
 
       if (functionObj.events) {
         functionObj.events.forEach(event => {
+          let enabled = true;
+          let appId;
           if (event === 'alexaSkill') {
-            const lambdaLogicalId = this.provider.naming
-              .getLambdaLogicalId(functionName);
-
-            const permissionTemplate = {
-              Type: 'AWS::Lambda::Permission',
-              Properties: {
-                FunctionName: {
-                  'Fn::GetAtt': [
-                    lambdaLogicalId,
-                    'Arn',
-                  ],
-                },
-                Action: 'lambda:InvokeFunction',
-                Principal: 'alexa-appkit.amazon.com',
-              },
-            };
-
-            const lambdaPermissionLogicalId = this.provider.naming
-              .getLambdaAlexaSkillPermissionLogicalId(functionName);
-
-            const permissionCloudForamtionResource = {
-              [lambdaPermissionLogicalId]: permissionTemplate,
-            };
-
-            _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-              permissionCloudForamtionResource);
-          } else if (event.alexaSkill) {
+            const warningMessage = [
+              'Warning! You are using an old syntax for alexaSkill which doesn\'t',
+              ' restrict the invocation solely to your skill.',
+              ' Please refer to the documentation for additional information.',
+            ].join('');
+            this.serverless.cli.log(warningMessage);
+          } else if (typeof event.alexaSkill === 'string') {
+            appId = event.alexaSkill;
+          } else if (typeof event.alexaSkill === 'object') {
+            if (typeof event.alexaSkill.appId !== 'string') {
+              const errorMessage = [
+                `Missing "appId" property for alexaSkill event in function ${functionName}`,
+                ' The correct syntax is: appId: amzn1.ask.skill.xx-xx-xx-xx-xx',
+                ' OR an object with "appId" property.',
+                ' Please check the docs for more info.',
+              ].join('');
+              throw new this.serverless.classes.Error(errorMessage);
+            }
+            appId = event.alexaSkill.appId;
+            enabled = event.alexaSkill.enabled !== false;
```

Comment:
`enabled = event.alexaSkill.enabled;` - having the explicit not-equal check for false just obscures this and lessens readability.

Reply:
the explicit not-equal check here is because the `enabled` option is optional and defaults to true.
The following definition:
```yaml
- alexaSkill:
    appId: amzn1.ask.skill.yy-yy-yy-yy
```
should be considered enabled.

Reply:
Oh, ok. Maybe you can add a comment above the assignment, because it is not obvious without knowing exactly this context 😄